### PR TITLE
Trim URIs if necessary, and test that.

### PR DESCRIPTION
**Pull request by mmeeks**
Change-Id: I4f8cef1fb05a6f8fa8f85010adbf8b0ccd8af9e9

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: test/integration-http-server.cpp [hunk 1]
```diff
@@ -168,7 +169,35 @@ void HTTPServerTest::testCoolPostPoco()
...
+            {
+                if (split[j] == "'self'" || split[j] == "data:" || split[j] == "blob:")
+                    continue;
+
+                Poco::URI uri(split[j]);
+                LOG_TST("URL - " << split[j]);
+                LOK_ASSERT_EQUAL(std::string(""), uri.getUserInfo());
+                LOK_ASSERT(uri.getPath() == std::string("") ||
+                           uri.getPath() == std::string("*"));
+                LOK_ASSERT_EQUAL(std::string(""), uri.getQuery());
```
Comment (Ashod) [inline test/integration-http-server.cpp:196]: `std::string()` is sufficient.

File: test/integration-http-server.cpp [hunk 2]
```diff
@@ -168,7 +169,35 @@ void HTTPServerTest::testCoolPostPoco()
...
     LOK_ASSERT(html.find(form["access_token"]) != std::string::npos);
+    LOK_ASSERT(html.find(form["buy_product"]) != std::string::npos);
     LOK_ASSERT(html.find(_uri.getHost()) != std::string::npos);
+
+    std::string csp = response["Content-Security-Policy"];
+    StringVector lines = StringVector::tokenize(csp, ';');
+    LOG_TST("CSP - " << csp << " tokens " << lines.size());
+    for (size_t i = 0; i < lines.size(); ++i)
+    {
+        if(lines.startsWith(i, " connect-src") ||
```
Comment (Ashod) [inline test/integration-http-server.cpp:180]: Are these necessarily to start with a space? Perhaps we should allow for no spaces?

**Review by Ashod — APPROVED** [review]
Thanks, @mmeeks. Looks good.

File: test/integration-http-server.cpp (hunk 2 above)
Comment (mmeeks) [reply test/integration-http-server.cpp:180]: If C++ had a nice 'chomp' function that wasn't an explosion of evil - I'd use it; but as-is, this works fine for the test =)

**Comment by mmeeks** [comment]
@timar would love this in the stable branch too.

**Review by caolanm — APPROVED** [review]
looks sane to me

**Pull request merged**